sfxclient/httpsink.go

```diff
@@ -131,6 +135,24 @@ func toMT(mt datapoint.MetricType) com_signalfx_metrics_protobuf.MetricType {
 	panic(fmt.Sprintf("Unknown metric type: %d\n", mt))
 }
 
+var toECMap = map[event.Category]com_signalfx_metrics_protobuf.EventCategory{
+	event.USERDEFINED:      com_signalfx_metrics_protobuf.EventCategory_USER_DEFINED,
+	event.ALERT:            com_signalfx_metrics_protobuf.EventCategory_ALERT,
+	event.AUDIT:            com_signalfx_metrics_protobuf.EventCategory_AUDIT,
+	event.JOB:              com_signalfx_metrics_protobuf.EventCategory_JOB,
+	event.COLLECTD:         com_signalfx_metrics_protobuf.EventCategory_COLLECTD,
+	event.SERVICEDISCOVERY: com_signalfx_metrics_protobuf.EventCategory_SERVICE_DISCOVERY,
+	event.EXCEPTION:        com_signalfx_metrics_protobuf.EventCategory_EXCEPTION,
+}
+
+func toEC(ec event.Category) com_signalfx_metrics_protobuf.EventCategory {
```

Comment:
I removed the map entirely.  Instead there is a "toEC()" function that does validate whether a supplied event.Category matches a protobuf.Category, but it does it in two steps.  
1.  It checks whether there is a protobuf Enum for the integer
2. It returns a new protobufEnum using the event.Category integer.